core/trino-spi/src/main/java/io/trino/spi/connector/ConstraintEvaluator.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.spi.connector;
+
+import io.trino.spi.predicate.NullableValue;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+public interface ConstraintEvaluator
```

Comment:
some javadoc?

Reply:
i can give it a try ;)

Reply:
Maybe just:
```
Represents a constraint passed to {@link ConnectorSplitManager#getSplits}.
If {@link ConstraintEvaluator#isCandidate} returns false given actual partial column values binding, for given split, 
it means that split can safely be dropped without processing.
```

Reply:
i think i know why we do not pass `Constraint` into split manager -- we don't want to pass a TupleDomain so that connector doesn't want to intersect DynamicFilter's domain with Constraint's domain.

but then, the two classes: `Constraint` and `ConstraintEvaluator` have naturally similar names, and the second looks like the evaluator of the first. 

- i am not sure about the naming. maybe we can find a better name, like `TestableConstraint` or something
- should `Constraint` reuse this new class?

Reply:
cc @losipiuk ^